lib/puppet/functions/vault_lookup/lookup.rb

```diff
@@ -1,10 +1,12 @@
 Puppet::Functions.create_function(:'vault_lookup::lookup') do
   dispatch :lookup do
-    param 'String', :path
+    required_param 'String', :path
     optional_param 'String', :vault_url
+    optional_param 'Variant[Boolean, String]', :local_token
+    optional_param 'String', :local_token_file
   end
 
-  def lookup(path, vault_url = nil)
+  def lookup(path, vault_url = nil, local_token = false, local_token_file = '/etc/vault.token')
```

Comment:
@tuxmaster5000 have you considered in adding in support for [Hashicorp environment variables](https://www.vaultproject.io/docs/commands/#environment-variables)?

I'm not sure of your use case, but I think that adding support similar to what is already present for `VAULT_ADDR` might be a better approach than expanding the argument list here.

Reply:
As far I understood the link, the variable store the token itself. But my idea was to add an option the load the token from an file.

Reply:
I suggested following support for some of the Hashicorp environment variables because it seems like a good idea to follow some of the standard ways Hashicorp supports reading in the token. If we are going to add support for it as a method argument, it might be a good idea to add support reading it from the environment as well? There is already a pattern for reading in the vault_url from the `ENV`, and I think it would be good to extend that pattern.

Reply:
In this implementation, the code is setting a default location for the token file to be in `/etc/vault.token`, but the standard place is often [`~/.vault-token`](https://www.vaultproject.io/docs/commands/token-helper.html). Is this the reason for splitting `local_token` and `local_token_file`, so we can set that default location? I'm not sure we need to provide that default here, but I'd like to hear why the code is written that way.